.github/workflows/aws-prod.yml

```diff
@@ -2,64 +2,79 @@ on:
         aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
         aws-region: ${{ env.REGION }}
 
-    - name: Login to container registry
+    - name: Login to AWS ECR
       id: login-ecr
       uses: aws-actions/amazon-ecr-login@v1
 
-    - name: Build, tag, and push image to container registry
+    - name: Extract commit hash
+      id: vars
+      shell: bash
+      run: |
+        echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
+
+    - name: Build, tag, and push image to ECR
       id: build-image
       env:
+        SHA_TAG: ${{ steps.vars.outputs.sha_short }}
+        LATEST_TAG: latest
         ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
-        IMAGE_TAG: latest
-      run: |
-        # Build a docker container and
-        # push it to ECR so that it can
-        # be deployed to ECS.
-        git submodule update --init --recursive
-        docker build -t $ECR_REGISTRY/$SERVICE_NAME:$IMAGE_TAG .
-        docker push $ECR_REGISTRY/$SERVICE_NAME:$IMAGE_TAG
-        echo "::set-output name=image::$ECR_REGISTRY/$SERVICE_NAME:$IMAGE_TAG"
+      run: |  
+        # Build Docker containers and push them to ECR ${{ env.AWS_ECR_NAME }}
+        docker pull $ECR_REGISTRY/$AWS_ECR_NAME:$LATEST_TAG || true
+        docker build -t $AWS_ECR_NAME \
+                     -t $ECR_REGISTRY/$AWS_ECR_NAME:$SHA_TAG \
+                     -t $ECR_REGISTRY/$AWS_ECR_NAME:$LATEST_TAG \
+                     -t $ECR_REGISTRY/$AWS_ECR_NAME:$ENVIRONMENT_TAG \
+                     -f Dockerfile \
+                     .
 
-    - name: Download task definition
+        docker push $ECR_REGISTRY/$AWS_ECR_NAME --all-tags
+
+    - name: Preparing Helm values files
+      id: preparing-helm-values
+      env:
+        ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
       run: |
-        aws ecs describe-task-definition --task-definition ${SERVICE_NAME} --query taskDefinition > task-definition.json
- 
-    # At the moment we are always on "latest" tag anyway
-    # but going to render a new task definition anyway
-    - name: Render task definition
-      id: render
-      uses: aws-actions/amazon-ecs-render-task-definition@v1
-      with:
-        task-definition: task-definition.json
-        container-name: ${{ env.SERVICE_NAME }}
-        image: ${{ steps.build-image.outputs.image }}
+        sed -i 's/${ECR_REGISTRY}/'$(echo $ECR_REGISTRY)/ deploy/prod/values.yaml
 
-    - name: Deploy service
-      uses: aws-actions/amazon-ecs-deploy-task-definition@v1
+    - name: Deploying Service to Kubernetes with Helm
+      uses: bitovi/github-actions-deploy-eks-helm@v1.2.4
       with:
-        task-definition: ${{ steps.render.outputs.task-definition }}
-        service: ${{ env.SERVICE_NAME }}
-        cluster: ${{ env.SERVICE_NAME }}
-        wait-for-service-stability: true
+        values: image.repository=${{ steps.login-ecr.outputs.registry }}/${{ env.AWS_ECR_NAME }},image.tag=${{ steps.vars.outputs.sha_short }}
```

Comment:
Having this here, you shouldn't need step `preparing-helm-values` since setting the value here takes precedence.